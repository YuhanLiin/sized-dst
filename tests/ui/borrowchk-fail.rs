use sized_dst::*;

fn main() {
    {
        let n = 32;
        Dst::<dyn std::fmt::Debug, 16>::new(&n)
    };
}
