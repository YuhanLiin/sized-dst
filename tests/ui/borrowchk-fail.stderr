error[E0597]: `n` does not live long enough
 --> tests/ui/borrowchk-fail.rs:6:45
  |
5 |         let n = 32;
  |             - binding `n` declared here
6 |         Dst::<dyn std::fmt::Debug, 16>::new(&n)
  |         ------------------------------------^^-
  |         |                                   |
  |         |                                   borrowed value does not live long enough
  |         a temporary with access to the borrow is created here ...
7 |     };
  |     -- ... and the borrow might be used here, when that temporary is dropped and runs the `Drop` code for type `DstBase`
  |     |
  |     `n` dropped here while still borrowed
  |
help: consider adding semicolon after the expression so its temporaries are dropped sooner, before the local variables declared by the block are dropped
  |
6 |         Dst::<dyn std::fmt::Debug, 16>::new(&n);
  |                                                +
